questionGenerator.py

```diff
@@ -0,0 +1,291 @@
+import pandas as pd
+from utils import dataLoader, dataSaver, printAndLog
+from config import OpenAIBot
+import json
+import sys
+
+class QuestionData:
+    """
+    Represents a class that handles question data generation.
+
+    Args:
+        config (Config): The configuration object.
+        videoData (VideoData): The video data object.
+        topicModeller (TopicModeller): The topic modeller object.
+        load (bool, optional): Whether to load existing question data. Defaults to True.
+    """
+
+    def __init__(self, config, videoData, topicModeller, load=True):
+        self.config = config
+
+        self.clusteredTopics = getClusteredTopics(videoData, topicModeller)
+        self.dominantTopics = getDominantTopic(self.clusteredTopics)
+        self.relevantText, self.questionText = getTextAndQuestion(
+            self.dominantTopics, videoData
+        )
+        self.responseData = None
+
+        if load:
+            self.loadQuestionData()
+        else:
+            self.OpenAIChatBot = OpenAIBot(self.config)
+            self.getQuestionData()
+            config.questionTokenCount = self.OpenAIChatBot.tokenUsage
```

Comment:
are you trying to override the config Value? that's my reading from here. We usually don't override value related to configuration/env variable but just get then when running the process.

Reply:
For the token count? Yes, it's kind of being overridden here. 
The Config object is a persistent object that can store information throughout the pipeline, so that is why I was passing the token value to it as a way to store it. 
This does fall under Issue #17, where I would like to improve the way token usage might be tracked.

Reply:
@takposha , The problem I am trying to raise here is `Config` Class main purpose seems to be holding the `.env` variable. So you can look at this class as holder of a static info that only changes once before the process/script runs.

The movement you feed other variables that are irrelevant into this Object then it's not a Config Class anymore.

What you can do here is

1. Config Class only stores `.env` file values
2. Move all the static info variables to beginning `config.py` of the  of the  and grab those variable elsewhere by importing. Below are the variable that fits this. 
    1. `self.openAIParams`, 'self.captionsFolder`, `self.saveFolder`
    ```
     `config.py`
     CAPTIONS = "Captions"
      # when Importing do this say to `topicExtractor.py`
     from config import CAPTIONS
    # this solution works for single variable but I want to think through on the dict object, `openAPIParams` in similar context
    ```
3.  for variables `topicTokenCount`, `questionTokenCount` in the Config Class these as per my reading the current code it is only used that `QuestionData` and `TopicModeller`and only for logging purpose. So these need class variable for respective class and not part of the `Config` Class. If you see a purpose in future code otherthan logging, I feel you need pass the Class instance to grab the values.

Reply:
The code for Config.py was introduce in last PR https://github.com/tl-its-umich-edu/annoto-gai/pull/14. The PR was huge, so I couldn't raise the concern. This is one of the reason's why PR should be small or sizable for easy review.

Reply:
Hmm, I see. So what if I do the following:
I have three main classes: 
1. configVars for setting up environmental variables,
2. staticVars for storing all static variables that need to be passed,
3. trackingVars for storing any variables that are used to track and update stuff.

I put all of it together in one wrapper class to pass to all of the other functions (What do I call this, and I should maybe rename `config.py` to something else). Will this be a way to handle the different variables we are using in an effective manner?

Reply:
I will also create functions with the `TopicModeller` and `QuestionData` classes that return the token usage which can then be assigned to the variables that are tracking and logging these values. That will help with #17, so I will implement that in a smaller PR for that issue.

Reply:
`configVars` : This could be called as official Config class
`staticVars`: I don't know if you want to create class for static variable. These are simply be put in `contancts.py`
`trackingVars`: Separate it a different Object, don't know if you want a class for holding 2 variable. the topicTokenCount is part of `TopicModeller` and is available to `QuestionData`. So you could just create a simple dict object and pass it along. I will leave up to you.

Reply:
`callMaxLimit` should be part of `TopicModeller` and not part of `Config` class. just mentioning it here as well. I don't see this being used elsewhere

Reply:
Do we need to have the call limit ever set as an environmental value in the future, actually? I chose 5 as a default value, but if it doesn't need to be changed, I will move it into the actual classes where it is used only.

Reply:
Ideally yes, But I let it pass now. But I did notice it lasttime. I feel 3 should be good number than 5

Reply:
Noted. I will set it up as a fixed value in the class for now, and make an environmental value later if needed.